### ISSUE TITLE
Support editor folding of TLA+ translation of PlusCal algorithm

### ISSUE THREAD

**Issue by lemmy** [tlaplus/tlaplus#329]
Folding is currently implemented for steps of a TLA+ proof.  The same functionality should be added for the TLA+ translation of a PlusCal algorithm.  Folding support for boxed comments would be nice to have too.

[Folding in Eclipse Text Editors](https://www.eclipse.org/articles/Article-Folding-in-Eclipse-Text-Editors/folding.html)

**Comment by quaeler** [comment]
Addressed with 28f6e2c (code) and 0325cb9 (changelog update)

**Comment by lemmy** [comment]
If a multi-line comment is in-between the algorithm statement and the variables, collapsing the algorithm does not collapse the complete algorithm but only the multi-line comment.  A second folding is available for the rest of the PlusCal algorithm.

```tla
--algorithm OpenAddressing 
(******************************************************)
(* multi-line comment                                 *)          
(******************************************************)
{ variable table = [i \in 1..K |-> empty], 
           history = {};
 
...
```

**Comment by quaeler** [comment]
Can you attach the full `.tla` file (or point to it if its in the examples) containing this PlusCal?

**Comment by lemmy** [comment]
https://github.com/tlaplus/tlaplus/blob/master/tlatools/src/tlc2/tool/fp/OpenAddressing.tla

**Comment by quaeler** [comment]
Oh i see... not that we can unring this bell - but a comment-block that is inside another comment-block is generally considered illegal in languages in my experience - no?

**Comment by lemmy** [comment]
How difficult would it be to also support folding the TLA+ translation of the PlusCal algorithm (everything between BEGIN TRANSLATION and END TRANSLATION)?

**Comment by quaeler** [comment]
Should be doable - will implement!

**Comment by quaeler** [comment]
Addressed with 4898e3d

**Comment by quaeler** [comment]
- The expected comment block delineation for PlusCal algorithms has been documented in the help text.
- The GitHub Examples files which contain PlusCal have been ensured to obey the 'standard.'
- The gutter context menu now has a fold-all, and an expand-all, item.
- The Toolbox preferences now contain the ability to set a default folding preference for Block Comments, PlusCal algorithm block, and translated PlusCal TLA+ block.
